clients/include/testing_csx2dense.hpp

```diff
@@ -0,0 +1,396 @@
+{
+
+    int                  M        = argus.M;
+    int                  N        = argus.N;
+    int                  LD       = argus.lda;
+    hipsparseDirection_t dirA     = argus.dirA;
+    hipsparseIndexBase_t idx_base = argus.idx_base;
+
+    hipsparseStatus_t status;
+
+    std::unique_ptr<handle_struct> unique_ptr_handle(new handle_struct);
+    hipsparseHandle_t              handle = unique_ptr_handle->handle;
+
+    std::unique_ptr<descr_struct> unique_ptr_descr(new descr_struct);
+    hipsparseMatDescr_t           descr = unique_ptr_descr->descr;
+    CHECK_HIPSPARSE_ERROR(hipsparseSetMatIndexBase(descr, HIPSPARSE_INDEX_BASE_ZERO));
+
+    if(M <= 0 || N <= 0 || LD < M)
+    {
+#ifdef __HIP_PLATFORM_NVCC__
+        // Do not test args in cusparse
+        return HIPSPARSE_STATUS_SUCCESS;
+#endif
+        hipsparseStatus_t expected_status
+            = (((M == 0 && N >= 0) || (M >= 0 && N == 0)) && (LD >= M))
+                  ? HIPSPARSE_STATUS_SUCCESS
+                  : HIPSPARSE_STATUS_INVALID_VALUE;
+        status
+            = csx2dense(handle, M, N, descr, (const T*)nullptr, nullptr, nullptr, (T*)nullptr, LD);
+        verify_hipsparse_status(status,
+                                expected_status,
+                                (expected_status == HIPSPARSE_STATUS_SUCCESS)
+                                    ? "Error: call with zero sizes must be succesfull."
+                                    : "Error: An invalid size must be detected.");
+
+        return HIPSPARSE_STATUS_SUCCESS;
+    }
+
+    int              DIMDIR = (HIPSPARSE_DIRECTION_ROW == DIRA) ? M : N;
+    std::vector<T>   h_dense_val_ref(LD * N);
+    std::vector<T>   h_dense_val(LD * N);
+    std::vector<int> h_nnzPerRowColumn(DIMDIR);
+    std::vector<int> hd_nnzPerRowColumn(DIMDIR);
+    std::vector<int> h_nnzTotalDevHostPtr(1);
+    std::vector<int> hd_nnzTotalDevHostPtr(1);
+
+    //
+    // Create the dense matrix.
+    //
+    int  MN          = (dirA == HIPSPARSE_DIRECTION_ROW) ? M : N;
+    auto m_dense_val = hipsparse_unique_ptr{device_malloc(sizeof(T) * LD * N), device_free};
+    auto nnzPerRowColumn_managed
+        = hipsparse_unique_ptr{device_malloc(sizeof(int) * MN), device_free};
+    auto nnzTotalDevHostPtr_managed
+        = hipsparse_unique_ptr{device_malloc(sizeof(int) * 1), device_free};
+
+    T*   d_dense_val          = (T*)m_dense_val.get();
+    int* d_nnzPerRowColumn    = (int*)nnzPerRowColumn_managed.get();
+    int* d_nnzTotalDevHostPtr = (int*)nnzTotalDevHostPtr_managed.get();
+    if(!d_nnzPerRowColumn || !d_nnzTotalDevHostPtr || !d_dense_val)
+    {
+        verify_hipsparse_status_success(HIPSPARSE_STATUS_ALLOC_FAILED,
+                                        "!h_nnzPerRowColumn || !d_nnzPerRowColumn || !d_dense_val");
+        return HIPSPARSE_STATUS_ALLOC_FAILED;
+    }
+
+    //
+    // Initialize the entire allocated memory.
+    //
+    for(int i = 0; i < LD; ++i)
```

Comment:
SInce the values are always the same you can just collapse this into one for loop that iterates through the entire array.

Reply:
True, but question of taste? :) testing only..